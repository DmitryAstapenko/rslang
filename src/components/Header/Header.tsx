import React, { useState } from "react";
import { LinkContainer } from "react-router-bootstrap";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import { MAIN } from "../../constants/routes";
import Menu from "../Menu";
import UserBar from "../UserBar/UserBar";
import Score from "../Score";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 0,
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
    title: {
      flexGrow: 1,
      cursor: "pointer",
    },
  })
);

const Header: React.FunctionComponent = () => {
  const classes = useStyles();

  const [openMenu, setOpenMenu] = useState(false);

  const toggleDrawer = (open: boolean) => (
    event: React.KeyboardEvent | React.MouseEvent
  ) => {
    if (
      event.type === "keydown" &&
      ((event as React.KeyboardEvent).key === "Tab" ||
        (event as React.KeyboardEvent).key === "Shift")
    ) {
      return;
    }

    setOpenMenu(open);
  };

  return (
    <>
      <Menu open={openMenu} closeMenu={toggleDrawer} />
      <div className={classes.root}>
        <AppBar position="fixed">
          <Toolbar>
            <IconButton
              edge="start"
              className={classes.menuButton}
              color="inherit"
              aria-label="menu"
              onClick={toggleDrawer(true)}
            >
              <MenuIcon />
            </IconButton>
            <LinkContainer to={MAIN}>
              <Typography variant="h6" className={classes.title}>
                RSLang
              </Typography>
            </LinkContainer>
            <Score />
            <UserBar />
          </Toolbar>
        </AppBar>
      </div>
    </>
  );
};

export default Header;
