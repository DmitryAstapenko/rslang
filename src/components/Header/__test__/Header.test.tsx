import React from "react";
import Header from "../Header";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const setUp = () => shallow(<Header />);

describe("should render Header component", () => {
  let component:any;
  beforeEach(() => {
    component = setUp();
  });
  it("should contain div in Header", () => {
    const wrapper = component.find("div");
    expect(wrapper.length).toBe(1);
  });
  it("should render Header component", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
