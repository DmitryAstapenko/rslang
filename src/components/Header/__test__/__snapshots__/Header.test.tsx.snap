// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component should render Header component 1`] = `
<Fragment>
  <Menu
    closeMenu={[Function]}
    open={false}
  />
  <div
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(AppBar))
      position="fixed"
    >
      <WithStyles(ForwardRef(Toolbar))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="menu"
          className="makeStyles-menuButton-2"
          color="inherit"
          edge="start"
        >
          <Memo(ForwardRef(MenuIcon))
            onClick={[Function]}
          />
        </WithStyles(ForwardRef(IconButton))>
        <withRouter(LinkContainer)
          to="/"
        >
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-title-3"
            variant="h6"
          >
            RSLang
          </WithStyles(ForwardRef(Typography))>
        </withRouter(LinkContainer)>
        <UserBar />
      </WithStyles(ForwardRef(Toolbar))>
    </WithStyles(ForwardRef(AppBar))>
  </div>
</Fragment>
`;
